### PR TITLE
Wayland fixes

### PR DESCRIPTION
**Pull request by danigm**
https://phabricator.endlessm.com/T30889

### CODE REVIEW

File: eosclubhouse/clubhouse.py [hunk 1]
```diff
@@ -469,6 +470,10 @@ def _close_message(self, _button):
     def _animate(self, direction):
         display = self.get_screen().get_display()
         primary_monitor = display.get_primary_monitor()
+
+        if not primary_monitor:
+            primary_monitor = display.get_monitor_at_window(self.get_window())
```
Comment (manuq) [inline eosclubhouse/clubhouse.py:475]: :+1: . The code duplication is growing. I think is a good time to refactor the logic to obtain the monitor in a separate private method.

File: eosclubhouse/clubhouse.py [hunk 2]
```diff
@@ -518,6 +523,10 @@ def _place(self):
         x, y = 0, 0
         display = self.get_screen().get_display()
         primary_monitor = display.get_primary_monitor()
+
+        if not primary_monitor:
+            primary_monitor = display.get_monitor_at_window(self.get_window())
```
Comment (manuq) [inline eosclubhouse/clubhouse.py:528]: Exact same code as in `_animate`.

**Review by manuq — CHANGES_REQUESTED** [review]
Looks good. But can we discuss the second commit before merging? Are we going to use the 2 approaches for notifications? This forking path will be hard to test, hard to maintain, etc. If there is no way to do the "in app notification" in Wayland, we better use the extension notifications everywhere.

File: eosclubhouse/clubhouse.py [hunk 3]
```diff
@@ -62,6 +62,7 @@
 CLUBHOUSE_NAME = 'com.hack_computer.Clubhouse'
 CLUBHOUSE_PATH = '/com/hack_computer/Clubhouse'
 CLUBHOUSE_IFACE = CLUBHOUSE_NAME
+USE_INAPP_NOTIFICATIONS = False
```
Comment (manuq) [inline eosclubhouse/clubhouse.py:65]: Ah, just saw this. So is not detecting Wayland. Is using the extension always. :+1: then, good to go!

**Review by manuq — APPROVED** [review]

**Comment by danigm** [comment]
>  If there is no way to do the "in app notification" in Wayland, we better use the extension notifications everywhere.

Yes, that was my idea, but I didn't want to remove all the `InAppNotification` code. So this solution is to use the shell notifications everywhere, but with the posibility to recover the in-app notifications in the future.

I don't know, maybe this is too conservative. We can also remove the `InAppNotification` class and all the duplicated methods.

**Pull request merged**